src/integrationTest/java/org/opensearch/security/SecurityConfigurationBootstrapTests.java

```diff
@@ -0,0 +1,102 @@
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.opensearch.client.Client;
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.test.framework.TestSecurityConfig.User;
+import org.opensearch.test.framework.cluster.ClusterManager;
+import org.opensearch.test.framework.cluster.ContextHeaderDecoratorClient;
+import org.opensearch.test.framework.cluster.LocalCluster;
+import org.opensearch.test.framework.cluster.TestRestClient;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_ROLES_ENABLED;
+import static org.opensearch.test.framework.TestSecurityConfig.Role.ALL_ACCESS;
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class SecurityConfigurationBootstrapTests {
+
+    private final static Path configurationFolder = ConfigurationFiles.createConfigurationDirectory();
+
+    private static final User USER_ADMIN = new User("admin").roles(ALL_ACCESS);
+
+    @ClassRule
+    public static LocalCluster cluster = new LocalCluster.Builder().clusterManager(ClusterManager.THREE_CLUSTER_MANAGERS)
+        .loadConfigurationIntoIndex(false)
+        .defaultConfigurationInitDirectory(configurationFolder.toString())
+        .nodeSettings(
+            Map.of(
+                SECURITY_RESTAPI_ROLES_ENABLED,
+                List.of("user_" + USER_ADMIN.getName() + "__" + ALL_ACCESS.getName()),
+                SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX,
+                true
+            )
+        )
+        .build();
+
+    @BeforeClass
+    public static void initData() {
+
+        Client client = new ContextHeaderDecoratorClient(
+            cluster.getInternalNodeClient(),
+            Map.of(ConfigConstants.OPENDISTRO_SECURITY_CONF_REQUEST_HEADER, "true")
+        );
+        Runnable r = new Runnable() {
+            public void run() {
+                long t = System.currentTimeMillis();
+                long end = t + 10000;
+                while (System.currentTimeMillis() < end) {
+                    cluster.triggerConfigurationReloadForSingleCType(client, CType.CONFIG, true);
+                    try {
+                        Thread.sleep(50);
+                    } catch (InterruptedException e) {
+                        break;
+                    }
+                }
+            }
+        };
+
+        new Thread(r).start();
+    }
+
+    @AfterClass
+    public static void cleanConfigurationDirectory() throws IOException {
+        FileUtils.deleteDirectory(configurationFolder.toFile());
+    }
+
+    @Test
+    public void shouldStillLoadSecurityConfigDuringBootstrapAndActiveConfigUpdateRequests() {
```

Comment:
How do you think this can be re-factor? The primary setup for this is in the method annotated with @BeforeClass that will start a bgThread and run config update requests in a loop while the cluster spins up. The test case will verify that the security index is fully initialized by running an authInfo request with a user.

This ensures that the security cache is not partially seeded with only information from `config.yml` and instead gets other config types like internal users as well.

Reply:
If you want to swap the ordering for Peter's comment you can move the labels so the Class rule is effectively the before and the action is then the looping behavior you have put together. To me this seems like the behavior you are really trying to test here so I think calling that the "action" would be appropriate. The verification would then be that nothing fails or is  unpopulated after that action.

Reply:
See this pull request [1] for a way this could be done

- [1] https://github.com/cwperks/security/pull/18